src/zepben/ewb/services/common/translator/base_proto2cim.py

```diff
@@ -23,10 +30,32 @@
 from zepben.ewb.services.common.base_service import BaseService
 
 
+TProtoToCimFunc = Callable[[Message, BaseService], Optional[IdentifiedObject]]
+P = ParamSpec("P")
+R = TypeVar("R")
+
+
+def add_to_network_or_none(func: TProtoToCimFunc) -> TProtoToCimFunc:
```

Comment:
should document that this should be used for any leaf classes of the hierarchy

Reply:
documented?